docs/release.md

```diff
@@ -0,0 +1,9 @@
+# Release guide
+
+We follow semantic versioning for our releases.  The release procedure is fairly simple:
+1. Update [the changelog](../CHANGELOG.md)
+2. The release proces:
+   - merge the develop branch into the release branch.
+   - merge the release branch into the master branch.
+     **Note:** as you're merging into the master branch this requires you to have administrative priviliges.  You can go ahead and merge without approval (providing the develop branch was green).
```

Comment:
Love your attitude for adding a documentation for the release strategy!

In essence you nailed it.  For some insights in how I bake a release (maybe cool to add as an example?) I'd like to add the following.

# Hands on example:

1. We write the changelog entires on every PR, or alternatively write it right before launching the release on the `develop` branch. The current Changelog shows some examples on how we like to write them. In essence the PR title (including the PR sequence number) should already be descriptive enough to use as a changelog entry. On larger releases we like to categorize them by type: bug, feature, chore, ...
2. Merge to the release branch. We follow semantic versioning so please consider what kind of release number will be used. When a new minor release is tagged, open a new release branch. We follow this format for naming the branch: `release/4.2` All successive 4.2.x releases will be merged to that branch.
3. Next, when a new release is created, the release branch is merged to `master`. When dealing with a backport release. Only merge the change to the older release branch, say: `release/3.9`. 
4. From the master branch (or in case of a backport release, on the release branch itself) we create a tag. Tags do not include the `v` prefix.
5. Finally in Github we copy paste the release notes from the changelo onto the newly created release tag.

Tying it all together for a new release:

 ```bash
$ git checkout develop
$ vi CHANGELOG.md
$ git commit -a
$ git push
$ git checkout -b release/4.2
$ git push origin release/4.2
$ git checkout master
$ git merge release/4.2
$ git push
$ git tag 4.2.0
$ git push --tags
```

Reply:
❤️ this.  I've adjusted my text slightly based on some points in yours that were not entirely clear (or simply absent).  And then added your text below as the practical example.